api/app/src/command/medical/document/document-download.ts

```diff
@@ -1,16 +1,24 @@
-import { makeS3Client } from "../../../external/aws/s3";
+import { makeLambdaClient } from "../../../external/aws/lambda";
 import { Config } from "../../../shared/config";
 
-const s3client = makeS3Client();
+const lambdaClient = makeLambdaClient();
 
-export const downloadDocument = async ({ fileName }: { fileName: string }): Promise<string> => {
-  const seconds = 60;
+export const downloadDocument = async ({
+  fileName,
+  conversionType,
+}: {
+  fileName: string;
+  conversionType?: string;
+}): Promise<string> => {
+  const result = await lambdaClient
+    .invoke({
+      FunctionName: Config.getConvertDocLambdaName() ?? "",
+      InvocationType: "RequestResponse",
+      Payload: JSON.stringify({ fileName, conversionType }),
+    })
+    .promise();
 
-  const url = s3client.getSignedUrl("getObject", {
-    Bucket: Config.getMedicalDocumentsBucketName(),
-    Key: fileName,
-    Expires: seconds,
-  });
+  if (result.Payload === undefined) throw new Error("Payload is undefined");
 
-  return url;
+  return result.Payload.toString();
```

Comment:
Any particular reason not to keep the signed URL logic here so we don't need to call a lambda when there's no conversion requested?

Reply:
Yup, this was my main comment as well - we don't need to be calling the lambda to do a conversion if there's no conversion **required**.

Note that even if a conversion is requested, ideally we don't have the overhead of calling the lambda if the converted file already exists.